frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/capacity-and-nodes-step/stretch-cluster.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  FormGroup,
+  Alert,
+  Checkbox,
+  Select,
+  SelectVariant,
+  SelectOption,
+  TextContent,
+  TextVariants,
+  Text,
+  SelectProps,
+} from '@patternfly/react-core';
+import { FieldLevelHelp } from '@console/internal/components/utils';
+import {
+  createNodeZoneMap,
+  isValidStretchClusterTopology,
+  ZoneNodeMap,
+} from '../../../../utils/create-storage-system';
+import { arbiterText } from '../../../../constants';
+import { WizardNodeState, WizardState } from '../../reducer';
+import { AdvancedSubscription } from '../../advanced-subscription/advanced-subscription';
+
+const EnableArbiterLabel: React.FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="ocs-enable-arbiter-label">
+      <span className="ocs-enable-arbiter-label__title--padding">
+        {t('ceph-storage-plugin~Enable arbiter')}
+      </span>
+      <AdvancedSubscription />
+    </div>
+  );
+};
+
+const HelperText: React.FC = () => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <TextContent>
+        <Text component={TextVariants.small}>
+          {t(
+            'ceph-storage-plugin~To support high availability when two data centers can be used, enable arbiter to get the valid quorum between two data centers.',
+          )}
+        </Text>
+      </TextContent>
+      <Alert
+        aria-label={t('ceph-storage-plugin~Arbiter minimum requirements')}
+        title={t('ceph-storage-plugin~Arbiter minimum requirements')}
+        variant="info"
+        isInline
+      >
+        {arbiterText(t)}
+      </Alert>
+    </>
+  );
+};
+
+export const StretchCluster: React.FC<StretchClusterProps> = ({
+  onSelect,
+  nodes,
+  arbiterLocation,
+}) => {
+  const { t } = useTranslation();
+  const [isChecked, setIsChecked] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const options: SelectProps['children'] = [];
+  let isValidStretchCluster = false;
+
+  const zoneNodeMap: ZoneNodeMap = createNodeZoneMap(nodes);
+  const zones = Object.keys(zoneNodeMap);
+
+  if (zones.length) {
+    zones.forEach((zone) => options.push(<SelectOption key={zone} value={zone} />));
+    isValidStretchCluster = isValidStretchClusterTopology(zoneNodeMap);
+  }
```

Comment:
I am not seeing any significant issue, hence not memoizing it.
The list seems small and the component will re-render as being a child component if nodes get changed.